tests/report/common/tests.cpp

```diff
@@ -2,15 +2,292 @@
+
+    OE_CHECK_MSG(
+        oe_verify_report(enclave, remote_report, report_size, parsed_report),
+        "Failed to verify OE remote report. %s",
+        oe_result_str(result));
+#endif
+
+    OE_CHECK_MSG(
+        oe_get_collaterals_internal(
+            header->report,
+            header->report_size,
+            collaterals_buffer,
+            collaterals_buffer_size),
+        "Failed to get collaterals. %s",
+        oe_result_str(result));
+
+    result = OE_OK;
+done:
+    if (remote_report)
+        oe_free_report(remote_report);
+
+    OE_TRACE_INFO(
+        "Exit enclave call %s: %d(%s)\n",
+        __FUNCTION__,
+        result,
+        oe_result_str(result));
+
+    return result;
+}
+
+/**
+ * Verify the integrity of the report and its signature,
+ * with optional collateral data that is associated with the report.
+ *
+ * This function verifies that the report signature is valid. If the report is
+ * local, it verifies that it is correctly signed by the enclave
+ * platform. If the report is remote, it verifies that the signing authority is
+ * rooted to a trusted authority such as the enclave platform manufacturer.
+ *
+ * @param enclave The instance of the enclave that will be used to
+ * verify a local report. For remote reports, this parameter can be NULL.
+ * @param report The buffer containing the report to verify.
+ * @param report_size The size of the **report** buffer.
+ * @param collaterals The collateral data that is associated with the report.
+ * @param collaterals_size The size of the **collaterals** buffer.
+ * @param parsed_report Optional **oe_report_t** structure to populate with the
+ * report properties in a standard format.
+ *
+ * @retval OE_OK The report was successfully created.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ *
+ */
+static oe_result_t oe_verify_report_with_collaterals(
+#ifndef OE_BUILD_ENCLAVE
+    oe_enclave_t* enclave,
+#endif
+    const uint8_t* report,
+    size_t report_size,
+    const uint8_t* collaterals,
+    size_t collaterals_size,
+    oe_datetime_t* input_validation_time,
+    oe_report_t* parsed_report)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    oe_report_t oe_report = {0};
+    oe_report_header_t* header = (oe_report_header_t*)report;
+
+    if (report == NULL)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    if (report_size == 0 || report_size > OE_MAX_REPORT_SIZE)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    // Ensure that the report is parseable before using the header.
+    OE_CHECK(oe_parse_report(report, report_size, &oe_report));
+
+    if (header->report_type == OE_REPORT_TYPE_SGX_REMOTE)
+    {
+#ifndef OE_BUILD_ENCLAVE
+        // Intialize the quote provider if we want to verify a remote quote.
+        // Note that we don't have the OE_USE_LIBSGX guard here since we don't
+        // need the sgx libraries to verify the quote. All we need is the quote
+        // provider.
+        OE_CHECK(oe_initialize_quote_provider());
```

Comment:
The quote provider shouldn't be needed for host-side quote verification. Will this be addressed in a follow-up PR?

Reply:
The host provider is still needed to get the revocation information (TCB, CRL) and the QE identity info.

Reply:
I see, so if you have collateral then the quote provider is not needed, correct?

Reply:
Correct.